lua/lazyvim/plugins/extras/lang/cmake.lua

```diff
@@ -0,0 +1,40 @@
+return {
+  {
+    "nvim-treesitter/nvim-treesitter",
+    opts = function(_, opts)
+      if type(opts.ensure_installed) == "table" then
+        vim.list_extend(opts.ensure_installed, { "cmake" })
+      end
+    end,
+  },
+  {
+    "jose-elias-alvarez/null-ls.nvim",
+    opts = function(_, opts)
+      local nls = require("null-ls")
+      opts.sources = opts.sources or {}
+      vim.list_extend(opts.sources, {
+        nls.builtins.diagnostics.cmake_lint,
+      })
+    end,
+    dependencies = {
```

Comment:
I noticed this in your other PRs as well. Out of curiosity, why do you put `mason` as a dependency to `null-ls` and not on its own? In my personal config I've been putting it on its own so far. Is there a particular reason? (Something i should know to also take notice in my configuration perhaps)?

Reply:
The rationale behind this decision is not based on any particular reason. It came from my observation of configurations on GitHub that I found to be a clever approach to specifying the desired configuration. Specifically, the `null-ls` plugin requires the presence of a binary, which in turn relied on the `mason` plugin for installation of that binary.

Reply:
Yes, please don't add this as a dependency. That's not needed